### PR TITLE
feat: Rework draft handling front-end

### PR DESCRIPTION
**Pull request by hamza221**
Fixes https://github.com/nextcloud/mail/issues/7077
Closes https://github.com/nextcloud/mail/issues/4768

- [x] UC1
- [x] UC2
- [ ] UC3 => will be solved in a separate PR
- [x] UC4
- [x] UC5
- [x] UC6

### CODE REVIEW

**Comment by JohannesGGE** [comment]
There is something wrong with the background job. 

### UC1
- local_messages still contains entry

### UC2
- works fine

### UC3
- local_messages entry is not converted and remains in db

### UC4
- works fine

### UC5
- works fine

### UC6
- UC1 -> works fine. local_messages entry seems to be deleted this time ('send_at' value is set)
- UC2 -> works fine
- UC3 -> local_messages up to date but not deleted. Draft in messages disappeared and no new updated version appears

**Comment by hamza221** [comment]
UC3 still missing

**Comment by ChristophWurst** [comment]
UC1: the drafts are stored locally. adding more text updates that draft. sending puts the message into the outbox. but the message is not sent.

**Comment by ChristophWurst** [comment]
UC2+UC6 work :heavy_check_mark:

**Comment by ChristophWurst** [comment]
> UC1: the drafts are stored locally. adding more text updates that draft. sending puts the message into the outbox. but the message is not sent.

I think the only thing missing is starting a timer with the undo timeout and triggering a send of the new outbox message.

On systems with reliably cron the message will be sent after ~5mins. but I think the frontend should trigger sending right after the undo timeout.

**Comment by hamza221** [comment]
@ChristophWurst what. do you think should be done for UC3? should we handle it in a separate issue ?

**Comment by hamza221** [comment]
The timer is already there but the message's sendAt was not updated correctly, Also I noticed that messages get enqueue when the draft is saved, I changed that too

File: src/store/outbox/actions.js [hunk 1]
```diff
@@ -170,7 +169,7 @@ export default {
 					logger.error('Could not delay-send message ' + message.id, { message })
 					reject(error)
 				}
-			}, UNDO_DELAY)
+			}, UNDO_DELAY + 10)
```
Comment (ChristophWurst) [inline src/store/outbox/actions.js:172]: for testing?

Comment (hamza221) [reply src/store/outbox/actions.js]: Yes :/ , I removed it :))

**Review by JohannesGGE — APPROVED** [review]
With https://github.com/nextcloud/mail/pull/8576 U3 works now.

I only noticed one small thing:
In UC5: When there is no change the composer doesn't close completly. The minimized composer stays. You have to click on close again on the minimized composer.

**Review by JohannesGGE — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Green CI. Time for a squash.

**Pull request merged**